File: model-client/src/main/java/org/modelix/model/util/pmap/CustomPMap.java
package org.modelix.model.util.pmap;

public interface CustomPMap<K, V> {
  CustomPMap<K, V> put(K key, V value);
  V get(K key);
  CustomPMap<K, V> remove(K key);
  Iterable<K> keys();
  Iterable<V> values();
  boolean containsKey(K key);
}

File: model-client/src/main/java/org/modelix/model/util/pmap/CustomPMap.kt
package org.modelix.model.util.pmap

interface CustomPMap<K, V> {
    fun put(key: K, value: V): CustomPMap<K, V>?
    operator fun get(key: K): V
    fun remove(key: K): CustomPMap<K, V>?
    fun keys(): Iterable<K>?
    fun values(): Iterable<V>?
    fun containsKey(key: K): Boolean
}
